app-backend/api/src/main/scala/campaign/CampaignPermissionRoutes.scala

```diff
@@ -0,0 +1,155 @@
+package com.rasterfoundry.api.campaign
+
+import com.rasterfoundry.akkautil._
+import com.rasterfoundry.database._
+import com.rasterfoundry.datamodel._
+
+import akka.http.scaladsl.server._
+import cats.effect.IO
+import cats.implicits._
+import de.heikoseeberger.akkahttpcirce.ErrorAccumulatingCirceSupport._
+import doobie._
+import doobie.implicits._
+import doobie.util.transactor.Transactor
+
+import java.util.UUID
+
+trait CampaignPermissionRoutes
+    extends CommonHandlers
+    with Directives
+    with Authentication {
+
+  val xa: Transactor[IO]
+
+  def listCampaignPermissions(campaignId: UUID): Route = authenticate { user =>
+    authorizeScope(
+      ScopedAction(Domain.Campaigns, Action.ReadPermissions, None),
+      user
+    ) {
+      authorizeAuthResultAsync {
+        CampaignDao
+          .authorized(
+            user,
+            ObjectType.Campaign,
+            campaignId,
+            ActionType.Edit
+          )
+          .transact(xa)
+          .unsafeToFuture
+      } {
+        complete {
+          CampaignDao
+            .getPermissions(campaignId)
+            .transact(xa)
+            .unsafeToFuture
+        }
+      }
+    }
+  }
+
+  def replaceCampaignPermissions(campaignId: UUID): Route = authenticate {
```

Comment:
Do we need to limit the number of campaign shares? From the `addCampaignPermission` route it looks like we do

Reply:
Woops I missed 

```
                  CampaignDao.isReplaceWithinScopedLimit
```

below :sweat_smile: